viewing_party/party.py

```diff
@@ -1,23 +1,139 @@
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
+
+from viewing_party.party import *
+from dataclasses import dataclass
+#from symbol import subscript
+from xml.dom import UserDataHandler
+
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dictionary = {}
+    if title and genre and rating:
+        movie_dictionary["title"] = title
+        movie_dictionary["genre"] = genre
+        movie_dictionary["rating"] = rating
+        return movie_dictionary
+
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+#"watched": [{"title": "{movie} "genre": "Horror", "rating": 3.5},
+          #  {"title": "Title A", "genre": "Horror", "rating": 3.5},
+           # {"title": "Title A", "genre": "Horror", "rating": 3.5}]
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == title:
+            del user_data["watchlist"][i]
+            user_data["watched"].append(title)
+            return user_data
+
+        else:
+            return user_data
+
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    average = 0.0
+    sum = 0.0
+    if len(user_data["watched"]) != 0:
+        for num in user_data["watched"]:
+            sum += num["rating"] 
+        average = sum /len(user_data["watched"])
+        return average
+    else: 
+        average = 0.0
+        return average
 
+def get_most_watched_genre(user_data):
+    genremostwatched = {}
+    if len(user_data["watched"]) < 1:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            if movie["genre"] in genremostwatched.keys():
+                genremostwatched[movie["genre"]] += 1
+            else:
+                genremostwatched[movie["genre"]] = 1

```

Comment:
nice job putting the guard clause at the top! Let's take out the `else` statement, though. That way our main code isn't so crammed and indented:

```suggestion
    for movie in user_data["watched"]:
        if movie["genre"] in genremostwatched.keys():
            genremostwatched[movie["genre"]] += 1
        else:
            genremostwatched[movie["genre"]] = 1
```